cpu/podman_container.py

```diff
@@ -0,0 +1,137 @@
+        self.cpu_quota = self.params.get("cpu_quota", default=50000)
+        self.cpu_period = self.params.get("cpu_period", default=100000)
+        self.workload_run_time = self.params.get("run_time", default=1000)
+
+    def Generate_docker_file(self):
+        """
+        Generating the docker image with ebizzy binary attached
+        """
+
+        self.log.info("Inside the Generate_docker_file---")
+        Docker_row_data = "FROM busybox\nCOPY ebizzy \
+                /usr/local/bin/ebizzy\nRUN chmod +x /usr/local/bin/ebizzy"
+        with open("Dockerfile", 'w') as file_obj:
+            file_obj.write(Docker_row_data)
+
+    def Build_docker_image(self):
+        """
+        Building the docker image with ebizzy workload so that the
+        podman container will fetch the ebizzy image at run time
+        """
+        cmd = "podman build -t custom-busybox-ebizzy ."
+        process.system(cmd,
+                       ignore_status=True, shell=False, sudo=True)
+
+    def run_container(self):
+        """
+        Running the podman container and attaching ebizzy workload
+        """
+        workload_path = "/usr/local/bin/ebizzy"
+        cmd = "podman run --detach --cpu-period=%s --cpu-quota=%s \
+                custom-busybox-ebizzy %s -S %s" % (self.cpu_period,
+                                                   self.cpu_quota,
+                                                   workload_path,
+                                                   self.workload_run_time)
+
+        process.system(cmd,
+                       ignore_status=True, shell=False, sudo=True)
+
+    def log_generater(self, smt_state):
+        """
+        Capture the podman stats
+        :param smt_state: passing the smt state value to capture the logs
+        for same state.
+        """
+        podman_dir = self.logdir + "/podman"
+        if not os.path.isdir(podman_dir):
+            os.mkdir(podman_dir)
+
+        log_dir = podman_dir + "/" + str(smt_state) + ".txt"
+        cmd = "nohup podman stats &> %s &" % (log_dir)
+        process.run(cmd, shell=True)
+        self.log.info("Waiting for some time dump the podman stats to file")
+        time.sleep(30)
+        self.log.info("Killing the podman stats command PID")
+        cmd = 'pgrep -f "podman stats"'
+        cmd_output = process.run(cmd, shell=True)
+        process_pid = cmd_output.stdout
+        pid_decode = process_pid.decode('utf-8')
+        process_pid = pid_decode.strip("\n")
+        cmd = "kill %s" % (process_pid)
+        process.run(cmd, shell=True)
+
+    def test(self):
+        """
+        Here we are performing below four operations,
+        1. Generate_docker_file
+        2. Build_docker_image
+        3. run_container with ebizzy workload.
+        4. Capturing the podman stats log
+        """
+        self.Generate_docker_file()
+        self.Build_docker_image()
+        self.run_container()
+        cpu_controller = ["2", "4", "6", "on", "off"]
```

Comment:
is this something static?
can we push this into yaml?

Reply:
@Naresh-ibm Currently these **smt** mode combinations are static for this test case. So currently we don't need the .**yaml** file for this **SMT** mode.

Reply:
ok